sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java

```diff
@@ -96,6 +110,28 @@
             keyVaultUri = keyVaultUri + "/";
         }
         this.keyVaultUrl = keyVaultUri;
+        String dnsSuffix = Optional.of(keyVaultUri)
+            .map(uri -> uri.split("\\.",2)[1])
+            .map(suffix -> suffix.substring(0, suffix.length()-1))
+            .get();
+        this.keyVaultBaseUri = HTTPS_PREFIX + dnsSuffix;
+        switch(keyVaultBaseUri)
+        {
+            case KEY_VAULT_BASE_URI_GLOBAL :
+                this.aadAuthenticationUrl = AAD_LOGIN_GLOBAL_URI;
+                break;
+            case KEY_VAULT_BASE_URI_CN :
+                this.aadAuthenticationUrl = AAD_LOGIN_CN_URI;
+                break;
+            case KEY_VAULT_BASE_URI_US :
+                this.aadAuthenticationUrl = AAD_LOGIN_US_URI;
+                break;
+            case KEY_VAULT_BASE_URI_DE:
+                this.aadAuthenticationUrl = AAD_LOGIN_DE_URI;
+                break;
+            default:
+                throw new IllegalArgumentException("Property of azure.keyvault.uri is illegal.");
+        }
```

Comment:
1. Write a method in `Constants`: `getAADLoginURIByKeyVaultBaseUri`. 
2. And rename `Constants` to `UriUtil`.
3. Write unit test for the `getAADLoginURIByKeyVaultBaseUri`.

Reply:
Done with point 1&2. 
Do we still need a unit test for getAADLoginURIByKeyVaultBaseUri given it's just a switch statement? Does the current ut in KeyVaultClientTest meet the requirements?